app/frontend/app/controllers/group.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
+import HasCurrentUser from '../mixins/has-current-user';
+/* global Messenger */
+
+export default Ember.Controller.extend(HasCurrentUser, {
+  // this flag is used so that the "Leave Group" button doesn't switch
+  // to "Join Group" when waiting for the server to respond.
+  loading: false,
+
+  coverUpload: Ember.Object.create(),
+  coverUrl: Ember.computed.any('coverUpload.croppedImage', 'model.coverImage'),
+  coverImageStyle: function() {
+    return "background-image: url(" + this.get('coverUrl') + ")";
+  }.property('coverUrl'),
+
+  currentMember: function() {
```

Comment:
This shouldn't depend on us having a complete list of all members of a group on the client. There can be groups with a very large number of members.